test/androidTest/java/org/thoughtcrime/securesms/espresso/ConversationHelper.java

```diff
@@ -89,6 +93,36 @@ public ConversationHelper sendReaction(int position, int emoji) {
         return this;
     }
 
+    public ConversationHelper sendReply(int position, String message) {
+
+        this.unselectMessage();
+        this.selectMessage(position);
+
+        onView(withId(R.id.menu_context_reply))
+                .perform(click());
+        onView(withId(R.id.lin_reply))
+                .perform(click());
+        onView(withId(R.id.custom_reply))
+                .perform(typeText(message))
+
+                .perform(android.support.test.espresso.action.ViewActions.pressKey(KeyEvent.KEYCODE_ENTER))
+                .perform(replaceText(""))
+                .perform(android.support.test.espresso.action.ViewActions.pressKey(KeyEvent.KEYCODE_BACK))
```

Comment:
Could you explain what these `KEYCODE_BACK` events are doing?

Reply:
1st keycode back exits the keyboard, 2nd exits reply mode. This is the only way to exit reply mode currently that i know of.